src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,152 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+
+
+/**
+ * Storage class is used to interact with the File System and gather the save data from the last use session.
+ */
+public class Storage {
+
+    private static final Pattern pattern = Pattern.compile("^\\[(.)]\\s\\[(.)]\\s(.*?)(?:\\s/..\\s(.*))?$");
+
+    private String currentDir = System.getProperty("user.dir");
+    private Path dataDir = Paths.get(currentDir, "src", "main", "data");
+    private Path fileDir = Paths.get(currentDir, "src", "main", "data", "list.txt");
+    /**
+     * Checks if the /data directory exists.
+     *
+     * @return True if it exists, False otherwise.
+     */
+    private boolean dirExists() {
+        return Files.exists(dataDir);
+    }
+
+    /**
+     * Checks if the /data/list.txt file exists.
+     *
+     * @return True if it exists, False otherwise.
+     */
+    private boolean fileExists() {
+        return Files.exists(fileDir);
+    }
+
+    /**
+     * Private method to create the list.txt file in the /data directory.
+     *
+     * @return list.txt File
+     */
+    private File createFile() {
+        File newFile = new File(dataDir.toString(), "list.txt");
+        System.out.println("Creating File...");
+        try {
+            newFile.createNewFile();
+            System.out.println("Save file created at: " + this.fileDir);
+            return newFile;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Private method that gets the list.txt file if present, else a new
```

Comment:
The comments seems good and it would  be better if it could start with verbs instead following the Coding Standard